database/scrapers/faq_script.ts

```diff
@@ -0,0 +1,209 @@
+import { DateTime } from "luxon";
+import { Readable } from "node:stream";
+
+import { BaseScraperModule } from "#commands/db_scrape";
+import GuideArticle from "#models/guide_article";
+import GuideQuestion from "#models/guide_question";
+import FilesService from "#services/files_service";
+
+interface GuideArticlesOld {
+  data: {
+    id: number;
+    name: string;
+    cover: string;
+    short_description: string;
+    description: string | null;
+    order: number;
+    questions: number[];
+  }[];
+}
+
+interface GuideQuestionsOld {
+  data: {
+    id: number;
+    status: string;
+    date_created: string;
+    date_updated: string;
+    question: string;
+    answer: string;
+    type: number;
+  }[];
+}
+
+interface PivotTable {
+  data: {
+    id: number;
+    FAQ_Types_id: number | null;
+    FAQ_id: number;
+    sort: number | null;
+  }[];
+}
+
+interface ImageMetadata {
+  data: {
+    filename_disk: string | null;
+    filename_download: string | null;
+  };
+}
+
+async function uploadImage(imageUrl: string) {
+  const filesService = new FilesService();
+
+  const [fetchedImage, fetchedImageMetadata] = await Promise.all([
+    fetch(`https://admin.topwr.solvro.pl/assets/${imageUrl}`),
+    fetch(`https://admin.topwr.solvro.pl/files/${imageUrl}`),
+  ]);
+
+  if (!fetchedImage.ok) {
+    throw new Error(
+      `Failed to fetch the image. HTTP status: ${fetchedImage.status}`,
+    );
+  }
+
+  if (!fetchedImageMetadata.ok) {
+    throw new Error(
+      `Failed to fetch the image metadata. HTTP status: ${fetchedImageMetadata.status}`,
+    );
+  }
+
+  const imageMetadata = (await fetchedImageMetadata.json()) as ImageMetadata;
+  let extension: string | undefined = "";
+  if (imageMetadata.data.filename_disk !== null) {
+    if (imageMetadata.data.filename_disk.split(".").length > 1) {
+      extension = imageMetadata.data.filename_disk.split(".").pop();
+    }
+  }
+  if (extension === "") {
+    if (imageMetadata.data.filename_download !== null) {
+      if (imageMetadata.data.filename_download.split(".").length > 1) {
+        extension = imageMetadata.data.filename_download.split(".").pop();
+      }
+    }
+  }
+
+  if (extension === undefined || extension === "") {
+    throw new Error("Failed to determine image extension");
+  }
+
+  const stream = Readable.fromWeb(fetchedImage.body as ReadableStream);
+
+  try {
+    return await filesService.uploadStream(stream, extension);
+  } catch (error) {
+    throw new Error(`Failed to upload image.`, { cause: error });
+  }
```

Comment:
comment from previous review still unaddressed:
"Failed to upload image" adds no context.
If you can't add any context here, consider not catching the error and trying to add context one level higher